src/main/java/com/eppo/sdk/helpers/bandit/BanditEvaluator.java

```diff
@@ -0,0 +1,86 @@
+package com.eppo.sdk.helpers.bandit;
+
+import com.eppo.sdk.dto.EppoAttributes;
+import com.eppo.sdk.dto.EppoValue;
+import com.eppo.sdk.dto.ShardRange;
+import com.eppo.sdk.dto.Variation;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.nio.ByteBuffer;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+public class BanditEvaluator {
```

Comment:
Bandit evaluation is the template pattern with the strategy pattern.

The template is:
1. Get the relevant model (strategy) for weighing the actions
2. Use that model to weigh the actions
3. Shuffle the order of the actions
4. Translate the actions and their weights (`0.0`-`1.0`) to variations with shard ranges, in the shuffled order.